src/tools/execution-results.h

```diff
@@ -97,8 +109,17 @@ struct ExecutionResults {
           results[exp->name] = ret;
           // ignore the result if we hit an unreachable and returned no value
           if (ret.size() > 0) {
-            std::cout << "[fuzz-exec] note result: " << exp->name << " => "
-                      << ret << '\n';
+            std::cout << "[fuzz-exec] note result: " << exp->name << " => ";
+            auto resultType = func->sig.results;
+            if (resultType.isRef()) {
+              // Don't print reference values, as funcref(N) contains an index
+              // for example, which is not guaranteed to remain identical after
+              // optimizations.
+              std::cout << resultType;
+            } else {
+              std::cout << ret;
+            }
```

Comment:
Nit: We [used](https://github.com/WebAssembly/binaryen/pull/2451/files#diff-d2128595c03cac3cc010b8bbcf17973cc3583244dc6953a2755040825deb7082R73-R82) to do this here, which excluded reference types from both printing and comapring, but #3384 moved it to `areEqual`, which I think only handled comparison and not printing. Can we do that in the one place possibly?

Reply:
Hmm, good question but I don't see how to merge the two. The previous code handled comparing one run to another run inside the fuzzer. This code handles printing them out which allows comparing one run here to the results from another VM run in another process. I think we need both. And to unify them we'd need to have something like a result being not a Literal but either a Literal or a Type, and to vary over both...?

Reply:
I might be mistaken, but can't we execute this part only when the type is not a reference type? Which means, 1. We don't print it 2. We don't assign it to `ret`. Because we don't assign it to `ret`, we don't need to compare it later.
https://github.com/WebAssembly/binaryen/blob/cdac6b123b1231859dc0dfbcb88dfabe005855e3/src/tools/execution-results.h#L97-L102

Reply:
That could work, but a downside is that then we wouldn't compare it at all - that is, we wouldn't even compare the types?

Reply:
Hmm, you're right.